src/app/components/playlist/playlist.rs

```diff
@@ -16,6 +16,14 @@ pub trait PlaylistModel {
 
     fn current_song_id(&self) -> Option<String>;
 
+    fn playlist_song_ids(&self) -> Option<Vec<String>> {
```

Comment:
no need to add these methods to the trait :)

the trait methods are mostly here for more methods that the UI will need (eg the UI needs to know the current playing song to show it is active, or needs to be able to play a clicked song, etc)